### PR TITLE
Introduced a loop to create the buttons

### PR DESCRIPTION
**Pull request by jaykay99992**
Instead of adding each button manually with having the scaleFactor parameter as the only paramater changing, I introduced a loop that increases the scaleFactor in accordance to the existing code, without having to declare each button in a new line

### CODE REVIEW

File: jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java [hunk 1]
```diff
@@ -76,27 +76,20 @@ public PertPanel() {
...
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 1.5, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 2, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 3, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 4, null)));
-    group.add(rbmi);
+    for (double zoomLevel = 0.25; zoomLevel <= 4.0; zoomLevel += 0.25) {
```
Comment (ariouellette2000) [inline jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java]: The code could be made more flexible by extracting the start, end, and increment values of _zoomLevel_ into variables or constants.

File: jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java [hunk 2]
```diff
@@ -76,27 +76,20 @@ public PertPanel() {
...
+      if (zoomLevel == 1.0) {
+        m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, zoomLevel, null)));
+        rbmi.setSelected(true);
+        group.add(rbmi);
+      } else {
+        m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, zoomLevel, null)));
```
Comment (ariouellette2000) [inline jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java]: Simplify the conditional block by moving _m.add(rbmi..)_ and _group.add(rbmi)_ outside and after the if-else structure, eliminating redundant code and improving readability.
Ex: `if (zoomLevel == 0) { rbmi.setSelected(true); }` is the only condition to keep

**Comment by jaykay99992** [comment]
@ariouellette2000 So do i just decalare the JRadioButtonMenuItem in side the if-esle condition. then after that block i perform the m.add() and group.add() functions ? that is indeed less code and better readability

File: jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java (hunk 1 above)
Comment (phil3838) [inline jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java]: This refactoring improves the maintanability of the code by replacing the hardcoded JRadioButtonMenuItem instantiations with a 'for' loop that dynamically createes the menu items based on defined zoom levels.

**Review by phil3838 — APPROVED** [review]
Great modification

File: jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java [hunk 3]
```diff
@@ -76,27 +76,20 @@ public PertPanel() {
     JMenu m = new JMenu(labels.getString("view.zoomFactor.text"));
```
Comment (hichambel) [inline jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java]: Les noms de variables comme m, rbmi et pb ne sont pas très descriptifs. Il serait bon de choisir des noms plus explicites pour améliorer la lisibilité du code.

**Comment by jaykay99992** [comment]
@ariouellette2000 @hichambel Je viens de modifier le code pour correspondre à tous les commentaires de cette conversation. Est-ce correct ainsi ? Puis-je procéder à la fusion ?

**Comment by jaykay9999** [comment]
code fusionné

**Pull request merged**